internal/content_update_policy/default_content_update_policy_resource.go

```diff
@@ -0,0 +1,533 @@
+var (
+	_ resource.Resource                   = &defaultContentUpdatePolicyResource{}
+	_ resource.ResourceWithConfigure      = &defaultContentUpdatePolicyResource{}
+	_ resource.ResourceWithImportState    = &defaultContentUpdatePolicyResource{}
+	_ resource.ResourceWithValidateConfig = &defaultContentUpdatePolicyResource{}
+)
+
+// NewDefaultContentUpdatePolicyResource is a helper function to simplify the provider implementation.
+func NewDefaultContentUpdatePolicyResource() resource.Resource {
+	return &defaultContentUpdatePolicyResource{}
+}
+
+// defaultContentUpdatePolicyResource is the resource implementation.
+type defaultContentUpdatePolicyResource struct {
+	client *client.CrowdStrikeAPISpecification
+}
+
+// defaultContentUpdatePolicyResourceModel is the resource model.
+type defaultContentUpdatePolicyResourceModel struct {
+	ID                      types.String `tfsdk:"id"`
+	PlatformName            types.String `tfsdk:"platform_name"`
+	SensorOperations        types.Object `tfsdk:"sensor_operations"`
+	SystemCritical          types.Object `tfsdk:"system_critical"`
+	VulnerabilityManagement types.Object `tfsdk:"vulnerability_management"`
+	RapidResponse           types.Object `tfsdk:"rapid_response"`
+	LastUpdated             types.String `tfsdk:"last_updated"`
+
+	settings *contentUpdatePolicySettings `tfsdk:"-"`
+}
+
+
+
+// extract extracts the Go values from their terraform wrapped values.
+func (d *defaultContentUpdatePolicyResourceModel) extract(ctx context.Context) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	d.settings, diags = extractRingAssignments(
+		ctx,
+		d.SensorOperations,
+		d.SystemCritical,
+		d.VulnerabilityManagement,
+		d.RapidResponse,
+	)
+
+	return diags
+}
+
+// wrap transforms Go values to their terraform wrapped values.
+func (d *defaultContentUpdatePolicyResourceModel) wrap(
+	ctx context.Context,
+	policy models.ContentUpdatePolicyV1,
+) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	d.ID = types.StringValue(*policy.ID)
+
+	if d.PlatformName.IsNull() {
+		d.PlatformName = types.StringValue(*policy.PlatformName)
+	}
+
+	if !strings.EqualFold(d.PlatformName.ValueString(), *policy.PlatformName) {
+		diags.AddError(
+			"Mismatch platform_name",
+			fmt.Sprintf(
+				"The api returned the following platform_name: %s for default content update policy: %s, the terraform config has a platform_name value of %s. This should not be possible, if you imported this resource ensure you updated the platform_name to the correct value in your terraform config.\n\nIf you believe there is a bug in the provider or need help please let us know by opening a github issue here: https://github.com/CrowdStrike/terraform-provider-crowdstrike/issues",
+				*policy.PlatformName,
+				d.ID,
+				d.PlatformName.ValueString(),
+			),
+		)
+	}
+
+	d.SensorOperations, d.SystemCritical, d.VulnerabilityManagement, d.RapidResponse, diags = populateRingAssignments(ctx, policy)
```

Comment:
when you consolidated the logic you didn't update the wrap method of content_update_policy.go